src/rollup/plugins/externals.ts

```diff
@@ -269,8 +280,14 @@ export function externals(opts: NodeExternalsOptions): Plugin {
           return;
         }
         const src = resolve(opts.traceOptions.base, file);
-        const { pkgName, subpath } = parseNodeModulePath(file);
-        const dst = resolve(opts.outDir, `node_modules/${pkgName + subpath}`);
+        const { pkgName, subpath, baseDir } = parseNodeModulePath(file);
+        const version = await getPackageJson(resolve(baseDir, pkgName)).then(
+          (r) => r.version
+        );
+        const fullName = excludeOptimization.has(pkgName)
+          ? `${pkgName}@${version}`
+          : pkgName;
+        const dst = resolve(opts.outDir, `node_modules/${fullName + subpath}`);
```

Comment:
This would be a nice idea to track additional versions and copy them but they need to be close to the npm package that requires them (`node_modules/other-lib/node_modules/util`) in order to node_module's resolution working.

Reply:
Thanks for the guide <3
I've updated the PR to use nested node_modules. It still needs some tweaking and refactoring. I will be working on it a bit more to make the tests pass.